internal/agent/device/applications/provider/volume_test.go

```diff
@@ -0,0 +1,179 @@
+package provider
+
+import (
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/flightctl/flightctl/internal/agent/client"
+	"github.com/flightctl/flightctl/internal/agent/device/fileio"
+	"github.com/flightctl/flightctl/internal/api/common"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteComposeOverrideDiff(t *testing.T) {
+	require := require.New(t)
+	tests := []struct {
+		name     string
+		base     *common.ComposeSpec
+		patched  *common.ComposeSpec
+		renamed  map[string]string
+		expected string // expected YAML substring
+		written  bool
+	}{
+		{
+			name: "no changes",
+			base: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1": {External: true},
+				},
+				Services: map[string]common.ComposeService{
+					"svc1": {
+						Image:   "alpine",
+						Volumes: []string{"vol1:/data"},
+					},
+				},
+			},
+			patched: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1": {External: true},
+				},
+				Services: map[string]common.ComposeService{
+					"svc1": {
+						Image:   "alpine",
+						Volumes: []string{"vol1:/data"},
+					},
+				},
+			},
+			renamed: nil,
+			written: false,
+		},
+		{
+			name: "volume renamed",
+			base: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1": {External: true},
+				},
+			},
+			patched: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1-renamed": {External: true},
+				},
+			},
+			renamed: map[string]string{"vol1": "vol1-renamed"},
+			expected: `volumes:
+  vol1-renamed:
+    external: true`,
+			written: true,
+		},
+		{
+			name: "volume and service references renamed",
+			base: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1": {External: true},
+				},
+				Services: map[string]common.ComposeService{
+					"svc1": {
+						Image:   "nginx",
+						Volumes: []string{"vol1:/mnt"},
+					},
+				},
+			},
+			patched: &common.ComposeSpec{
+				Volumes: map[string]common.ComposeVolume{
+					"vol1-renamed": {External: true},
+				},
+				Services: map[string]common.ComposeService{
+					"svc1": {
+						Image:   "nginx",
+						Volumes: []string{"vol1-renamed:/mnt"},
+					},
+				},
+			},
+			renamed: map[string]string{"vol1": "vol1-renamed"},
+			expected: `services:
+  svc1:
+    image: nginx
+    volumes:
+    - vol1-renamed:/mnt
+volumes:
+  vol1-renamed:
+    external: true`,
```

Comment:
The override content should be just
```suggestion
			expected: `volumes:
  vol1:
    external:
      name: vol1-renamed`,
```

You do not need to modify the service definition or volume key. You just need to add the `name: <unique_name>` See [compose-spec docs](https://github.com/compose-spec/compose-spec/blob/main/07-volumes.md#name)